models/vrp.rb

```diff
@@ -358,6 +359,24 @@ def self.remove_unecessary_units(hash)
       hash[:shipments]&.each{ |s| s[:quantities]&.delete_if{ |quantity| needed_units.exclude? quantity[:unit_id] } }
     end
 
+    def self.remove_unecessary_relations(hash)
+      return hash unless hash[:relations].to_a.size.positive?
+
+      types_with_duration =
+        %w[minimum_day_lapse maximum_day_lapse
+           minimum_duration_lapse maximum_duration_lapse
+           vehicle_group_duration vehicle_group_duration_on_weeks
+           vehicle_group_duration_on_months vehicle_group_number]
+
+      hash[:relations].group_by{ |r| r[:type] }.each{ |type, relations_set|
+        next unless types_with_duration.include?(type)
+
+        relations_set.each{ |r|
+          hash[:relations].delete(r) unless r[:lapse]
+        }
+      }
```

Comment:
Why do we accept these incorrect relations in the first place? :thinking:  If we delete them like this then the user cannot know that they are ignored :/ 

Can't we use a `coerce_with` at API level so that if a relation needs "lapse" then json's with this relation but without "lapse" can be refused at the gate (which applies to other relations such as "shipments, order, etc.." needs `linked_ids`, there are others that render `linked_vehicle_ids` obligatory..)